lib/chore/queues/sqs/consumer.rb

```diff
@@ -66,46 +89,59 @@ def delay(item, backoff_calc)
 
         # Requests messages from SQS, and invokes the provided +&block+ over each one. Afterwards, the :on_fetch
         # hook will be invoked, per message
+        #
+        # @param [Proc] &handler Message handler, passed along by #consume
+        #
+        # @return [Array<Aws::SQS::Message>]
         def handle_messages(&block)
-          msg = queue.receive_messages(:limit => sqs_polling_amount, :attributes => [:receive_count])
+          msg = queue.receive_messages(:max_number_of_messages => sqs_polling_amount, :attribute_names => ['ApproximateReceiveCount'])
           messages = *msg
+
           messages.each do |message|
-            unless duplicate_message?(message.id, message.queue.url, queue_timeout)
-              block.call(message.handle, queue_name, queue_timeout, message.body, message.receive_count - 1)
+            unless duplicate_message?(message.message_id, message.queue_url, queue_timeout)
+              block.call(message.message_id, message.receipt_handle, queue_name, queue_timeout, message.body, message.attributes['ApproximateReceiveCount'].to_i - 1)
             end
-            Chore.run_hooks_for(:on_fetch, message.handle, message.body)
+            Chore.run_hooks_for(:on_fetch, message.receipt_handle, message.body)
           end
+
           messages
         end
 
-        # Retrieves the SQS queue with the given +name+. The method will cache the results to prevent round trips on
-        # subsequent calls. If <tt>reset_connection!</tt> has been called, this will result in the connection being
-        # re-initialized, as well as clear any cached results from prior calls
+        # Retrieves the SQS queue object. The method will cache the results to prevent round trips on subsequent calls
+        #
+        # If <tt>reset_connection!</tt> has been called, this will result in the connection being re-initialized,
+        # as well as clear any cached results from prior calls
+        #
+        # @return [Aws::SQS::Queue]
         def queue
           if !@sqs_last_connected || (@@reset_at && @@reset_at >= @sqs_last_connected)
-            AWS::Core::Http::ConnectionPool.pools.each do |p|
-              p.empty!
-            end
+            self.class.reset_connection!
             @sqs = nil
             @sqs_last_connected = Time.now
             @queue = nil
           end
-          @queue_url ||= sqs.queues.url_for(@queue_name)
-          @queue ||= sqs.queues[@queue_url]
+
+          @queue_url ||= sqs.get_queue_url(queue_name: @queue_name).queue_url
+          @queue ||= Aws::SQS::Queue.new(url: @queue_url)
```

Comment:
Looks like we should be passing in `client: sqs` here: https://github.com/aws/aws-sdk-ruby/blob/877d729fa4176891e68c77143b80a3ae6612b9e7/gems/aws-sdk-sqs/lib/aws-sdk-sqs/queue.rb#L16-L26

Example of how the Resource class builds queues: https://github.com/aws/aws-sdk-ruby/blob/877d729fa4176891e68c77143b80a3ae6612b9e7/gems/aws-sdk-sqs/lib/aws-sdk-sqs/resource.rb#L260-L269

Reply:
Good eye, thank you!